src/bases/finitebasis.jl

```diff
@@ -0,0 +1,60 @@
+import Base: length,
+    size
+
+###############
+# FiniteBasis #
+###############
+    # A FiniteBasis repesents the tensor product
+    # of an arbitrary number of fock bases, each of which is truncated 
+    # at a known, finite number of states. For example, imagine the 
+    # following fock bases and their tensor product:
+    #
+    # F_4 => {|0>, |1>, |2>, |3>}
+    # F_3 => {|0>, |1>, |2>}
+    # F_2 => {|0>, |1>} 
+    #
+    # F_4 x F_3 x F_2 => {|0,0,0>, |0,0,1>, |0,1,0>, |0,1,1>...|3,2,1>}
+    # 
+    # Note that the subscript number refers to basis *length*, not max
+    # excitation level.
+    #
+    # This tensor product basis, F_4 x F_3 x F_2, could be represented 
+    # by calling FiniteBasis(4, 3, 2). 
+
+    immutable FiniteBasis{S} <: AbstractBasis{S}
```

Comment:
After playing around with `NTuple`s more, I see I was off-base in what my assumptions were about how they worked. This is actually great, I've never messed with these before but you can be sure they're going to see a lot of use over in `QuDirac` as well...

My comment before about `NTuple`s not being suitable for non-uniform factor bases is invalid - I'll make the change this weekend. 

Also, I don't think there is anything to worry about in regards to tuple performance - they're more compact than normal arrays and their immutability gives the compiler a slight advantage in some cases (of course, there are many instances where mutable arrays *should* be used, but I think that `NTuple` really is the right way to go here).